### PR TITLE
[expo-contact][android] Fix `updateContactAsync `

### LINKED ISSUE #8997: [Contacts] Contacts.updateContactAsync creates new contact on android
### Summary of Issue (just a few sentences)
Contacts.updateContactAsync creates new contact on android instead of updating the existing one

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
android standalone

### Reproducible Demo
can't give a demo because you can only get write permissions on standalone.
but i guess the following code will do:

### Steps to Reproduce
1. run the following code in a standalone android app with WRITE_CONTACT and READ_CONTACT permissions:
`const contacts  = (await Contacts.getContactsAsync()).data;`
`const firstContact = contacts[0];`
`firstContact.firstName = firstContact.firstName + 'test expo';`
`await Contacts.updateContactAsync(firstContact)`;

2. go to your contacts app and search for "test expo".
3. search for the the real first name of the contact you found (the first name without the "test expo").
4. you will see both the original contact and a contact with the "test expo" added to its name.

### Expected Behavior vs Actual Behavior

Expected Behavior:
Contacts.updateContactAsync will update and exisitng contact.

Actual Behavior:
Contacts.updateContactAsync creates a new contact.

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes https://github.com/expo/expo/issues/8997.

# How

- reimplements `updateContactAsync` on iOS.
- rewrites test-suite.
- reformats android code.

# Test Plan

- https://snack.expo.io/@lukaszkosmaty/expo-contacts---fix-updatecontactasync-demo ✅

# Changelog

- Fixed `updateContactAsync` creating a new contact on Android.

### CODE REVIEW

File: packages/expo-contacts/ios/EXContacts/EXContacts.m [hunk 1]
```diff
@@ -314,8 +314,9 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
     if (!contact) return;
     [self _mutateContact:contact withData:updates resolver:resolve rejecter:reject];
     [saveRequest updateContact:contact];
-    [EXContacts executeSaveRequest:saveRequest contactStore:contactStore resolver:resolve rejecter:reject];
-}
+    if ([EXContacts executeSaveRequest:saveRequest contactStore:contactStore resolver:nil rejecter:reject]) {
+        resolve(contact.identifier);
```
Comment (tsapeta) [inline packages/expo-contacts/ios/EXContacts/EXContacts.m:318]: Shouldn't this be included in changelog as well? 🤔

**Review by tsapeta — APPROVED** [review]

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/BaseModel.java [hunk 2]
```diff
@@ -12,173 +12,191 @@
...
+  public Bundle getMap() {
+    return map;
+  }
+
+  protected String getLabelFromCursor(Cursor cursor) {
+    switch (cursor.getInt(cursor.getColumnIndex(EXColumns.TYPE))) {
+      case EXColumns.TYPE_CUSTOM:
+        final String label = cursor.getString(cursor.getColumnIndex(EXColumns.LABEL));
+        return label != null ? label : "unknown";
+      default:
```
Comment (mczernek) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/BaseModel.java:136]: Are there any changes apart from formatting 😅 ?

**Review by mczernek — APPROVED** [review]
Looks good! 💯

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/BaseModel.java (hunk 2 above)
Comment (lukmccall) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/BaseModel.java:136]: In this file nope. 
Please, check `ArrayList<ContentProviderOperation> toUpdateOperationList()` method and places where it is use ;)

**Pull request merged**